sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1718,7 +1718,7 @@ object SQLConf {
         "Data source V1 now silently drops partitionBy columns for non-file-format sources; " +
         "turning the flag on provides a way for these sources to see these partitionBy columns.")
       .booleanConf
-      .createWithDefault(false)
+      .createWithDefault(true)
```

Comment:
Could you describe more reasons in the PR description because Apache Spark 3.0 doesn't mean enabling all existing configurations?

Reply:
Sure - I added some clarification.